addOns/pscanrulesBeta/src/test/java/org/zaproxy/zap/extension/pscanrulesBeta/SubResourceIntegrityAttributeScanRuleUnitTest.java

```diff
@@ -117,6 +120,51 @@ void shouldNotRaiseAlertGivenIntegrityAttributeIsPresentInLinkElement()
         assertThat(alertsRaised.size(), equalTo(0));
     }
 
+    @Test
+    void shouldRaiseAlertGivenMissingIntegrityAttributeInLinkElement()
+            throws HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg =
+                buildMessage(
+                        "<html><head><link href=\"https://site53.example.net/style.css\"></head><body></body></html>");
+        // When
+        scanHttpResponseReceive(msg);
+        // Then
+        assertThat(alertsRaised.size(), equalTo(1));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"stylesheet", "STYLESHEET", "STYLEsheet", "alternate stylesheet"})
+    void shouldRaiseAlertGivenMissingIntegrityAttributeInLinkElementWithRelevantRel(String relVal)
+            throws HttpMalformedHeaderException {
+        // Given
+        HttpMessage msg =
+                buildMessage(
+                        "<html><head><link rel=\""
+                                + relVal
+                                + "\" href=\"https://site53.example.net/style.css\"></head><body></body></html>");
+        // When
+        scanHttpResponseReceive(msg);
+        // Then
+        assertThat(alertsRaised.size(), equalTo(1));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"dns-prefetch", "DNS-PREFETCH", "DNS-prefetch"})
+    void shouldNotRaiseAlertGivenMissingIntegrityAttributeInLinkElementWithIrrelevantRel(
```

Comment:
`shouldNotRaiseAlertGivenCanonicalAttributeIsPresentInLinkElement` should be removed superseded by this one or this one dropped and the other updated accordingly.